BookService/Tests/AuthorTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using Logic;
+using Logic.Models;
+using Moq;
+using Xunit;
+
+namespace Tests
+{
+    /// <summary>
+    /// Class for testing Authors
+    /// </summary>
+    public class AuthorTests : TestsBase
+    {
+        /// <summary>
+        /// Test for GetAuthorByIndex-method(Expected: exception)
+        /// </summary>
+        /// <param name="index">Index of the author</param>
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(100)]
+        public void Library_GetAuthorByIndex_Exception(int index)
+        {
+            // Arrage
+            InitLists();
```

Comment:
For most of the tests you don't need a lot values in the collection. You can easily initialize them directly in the test. To increase test readabilty. You will see what test data inserted, what was requested and assert result.